### PR TITLE
Disable source-highlighting for static build

### LINKED ISSUE #1487: Linking error compiling mips-unknown-linux-uclibc: cannot find -lsource-highlight
- crosstool-NG version : git master latest, commit hash: 77d75eb2
- Host distro and version: Arch Linux x86_64 (rolling release, up to date)
- Host kernel version: Linux 5.11.11-arch1-1 x86_64 GNU/Linux
- [.config](https://github.com/crosstool-ng/crosstool-ng/files/6306624/config.txt)
- [build.log.tar.gz](https://github.com/crosstool-ng/crosstool-ng/files/6306631/build.log.tar.gz)

Most relevant part of build.log:
```
[ALL  ]    /usr/bin/ld: gdbsupport/gdb-dlfcn.o: in function `gdb_dlopen(char const*)':
[ALL  ]    /home/dws/mips/.build/mips-unknown-linux-uclibc/src/gdb/gdb/gdbsupport/gdb-dlfcn.c:65: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
[ALL  ]    /usr/bin/ld: cannot find -lsource-highlight
[ERROR]    collect2: error: ld returned 1 exit status
[ERROR]    make[3]: *** [Makefile:1908: gdb] Error 1
[ALL  ]    make[3]: Leaving directory '/home/dws/mips/.build/mips-unknown-linux-uclibc/build/build-gdb-cross/gdb'
[ERROR]    make[2]: *** [Makefile:9231: all-gdb] Error 2
[ALL  ]    make[2]: Leaving directory '/home/dws/mips/.build/mips-unknown-linux-uclibc/build/build-gdb-cross'
[ERROR]    make[1]: *** [Makefile:853: all] Error 2
[ALL  ]    make[1]: Leaving directory '/home/dws/mips/.build/mips-unknown-linux-uclibc/build/build-gdb-cross'
[ERROR]
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing cross-gdb'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@376]
[ERROR]  >>        called from: do_gdb_backend[scripts/build/debug/300-gdb.sh@282]
[ERROR]  >>        called from: do_debug_gdb_build[scripts/build/debug/300-gdb.sh@59]
[ERROR]  >>        called from: do_debug[scripts/build/debug.sh@35]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@695]
```

### PR DESCRIPTION
**Pull request by QBos07**
Fixes #1487

### CODE REVIEW

File: scripts/build/debug/300-gdb.sh [hunk 1]
```diff
@@ -230,6 +230,7 @@ do_gdb_backend()
     if [ "${static}" = "y" ]; then
         cflags+=" -static"
         ldflags+=" -static"
+        extra_config+=("--without-source-highlight")
```
Comment (cpackham) [inline scripts/build/debug/300-gdb.sh:233]: I don't think this will work. The option is `--enable-source-highlight` so what we want is `--enable-source-highlight=no` or `--disable-source-highlight`. I think the configure script might even do the right thing if we point PKG_CONFIG_LIBDIR at the right place.

**Review by cpackham — CHANGES_REQUESTED** [review]

File: scripts/build/debug/300-gdb.sh (hunk 1 above)
Comment (QBos07) [reply scripts/build/debug/300-gdb.sh]: Fixed in 7ed7a6e

**Comment by cpackham** [comment]
Could you squash these into a single commit (use `git push -f` to stop github from complaining).

File: scripts/build/debug/300-gdb.sh [hunk 2]
```diff
@@ -230,11 +230,12 @@ do_gdb_backend()
     if [ "${static}" = "y" ]; then
         cflags+=" -static"
         ldflags+=" -static"
-        extra_config+=("--without-source-highlight")
+        extra_config+=("--disable-source-highlight") # static libsource-highlight is very rare
```
Comment (cpackham) [inline scripts/build/debug/300-gdb.sh:233]: I don't think the comment adds much..

File: scripts/build/debug/300-gdb.sh [hunk 3]
```diff
@@ -230,11 +230,12 @@ do_gdb_backend()
...
     fi
     if [ "${static_libstdcxx}" = "y" ]; then
         ldflags+=" -static-libgcc"
         ldflags+=" -static-libstdc++"
+        extra_config+=("--disable-source-highlight") # libsource-highlight uses exception handling and configure don't use dylibs with static exceptions
```
Comment (cpackham) [inline scripts/build/debug/300-gdb.sh:238]: Same here. The comment is just a bit too wordy to be useful.

**Review by cpackham — CHANGES_REQUESTED** [review]

**Comment by QBos07** [comment]
> Could you squash these into a single commit (use `git push -f` to stop github from complaining).

I try, but I wrote the comments with the web interface, because my internet connection is brocken. The initial connection to GitHub taking over 20 seconds, so I can't login.

**Comment by QBos07** [comment]
Better? xD

**Comment by cpackham** [comment]
> Better? xD

Much better thanks.

**Review by cpackham — APPROVED** [review]

**Comment by cpackham** [comment]
Merged in e3384374 I fixed a minor typo so github hasn't auto closed the PR
